### ISSUE TITLE
[NeuralNetworkApplication] Loading PredictFromModelProcess creates conflict with parsing utilities

### ISSUE THREAD

**Issue by danielandresarcones** [KratosMultiphysics/Kratos#9684]
**Description**
The NeuralNetworkApplication implements a process that loads an external neural network model to be used in the solver. If this process is initialized after another that parses an expression from a string (e.g. assign a variable from an expression), the parsing fails from that moment. When the process that includes the parsing is called, a "Kratos::Exception" is called and terminate is called recursively. The exception is catched exactly in the parsing of the expression. This doesn't occur if the neural network model is loaded in the solver or the analysis, only if it is called from the process. The process is predict_from_model_process.py and the initialization should not affect the parsing:
https://github.com/KratosMultiphysics/Kratos/blob/e7ed680545cea3ea299c0538f728a32be313d69a/applications/NeuralNetworkApplication/python_scripts/predict_from_model_process.py#L16-L35
This issue only happens with this specific process, which has no relation with the error. Right now it can be avoided by evaluating the model in the solver, but this will be a problem in the future. Any help or comments are welcomed.

**Scope**
- NeuralNetworkApplication

**To Reproduce**
1. Switch to branch`test-neuralnetwork-application`.
2. Compile the required applications (StructuralMechanicsApplication, FluidDynamicsApplication, MeshMovingApplication, LinearSolversApplication, CoSimulationApplication and NeuralNetworkApplication)
3. Keras and Tensorflow are needed.
4. Download the example from the [Examples repository](https://github.com/KratosMultiphysics/Examples/tree/test-neuralnetwork-application/neural_network/cosim/mok_fsi/surrogate_structure_legacy).
5. Run `run.sh`.
6. For successive runs, `run_without_training.sh` can be run instead.

**Expected behavior**
The error should not occur. An example of a working code achieved by not initializing the process can be found in [surrogate_fluid](https://github.com/KratosMultiphysics/Examples/tree/test-neuralnetwork-application/neural_network/cosim/mok_fsi/surrogate_fluid).

**Environment**
- OS: Ubuntu 20.04
- Branch: test-neuralnetwork-application
- Python 3.8.10
- Tensorflow 2.7.0
- Keras 2.7.0

**Comment by philbucher** [comment]
I am suspecting some weird behavior with the library we are using for a while
I am not sure what is the reason for those strange interactions, but I also encountered them :/
See #9689

**Comment by EduardGomezEscandell** [comment]
I solved my issues with function parsing by changing my locale. See my latest comments in #9689.

What does the expression that fails look like?

**Comment by danielandresarcones** [comment]
It is probably the same issue as in #9689 as my local is also set to Spanish. I will try that.
The expression has indeed decimal points, it is 2*0.06067*(1-cos(pi*t/10))*y*(1-y).

**Comment by EduardGomezEscandell** [comment]
Yeah, you can just try to use ` export LC_NUMERIC="en_US.UTF-8"` in the shell before running the program for a quick test.

Even if the expression didn't have decimal points, `assign_vector_by_modulus*` and similar processes add a constant with a decimal point in front of the expression to project it to each axis.

**Comment by danielandresarcones** [comment]
Thanks, it works fine now. I close the issue.
